Added Method Regression Analysis to exercises.qmd

jkc9886 opened the pull request:
One of the methods to visualize microbiome data in the OMA book chapters is regression charts, however there was no example or mention about it in the exercises chapter. 

I have added:

1. Regression Analysis under visualization heading after heatmaps (Line 1743)
2. Added description and steps for it.
3. Created an R code exercise to do Regression analysis using package ggplot2 and lm() function.

Please suggest if this could be added and if yes, what better could be done?

jkc9886 committed:
Method Regression Analysis to exercises.qmd

One of the methods to visualize data in the OMA book chapters is regression charts, however there was no example or mention about it in the exercises chapter. 

I have added:

1. Regression Analysis under visualization heading after heatmaps (Line 1743)
2. Added description and steps for it.
3. Added an R code to do Regression analysis using package ggplot2 and lm() function.

antagomir commented:
Thanks!

Regression is indeed a common statistical technique. 

The primary focus of OMA is to teach Bioconductor methods that support the modern multi-assay data containers, in particular the (Tree)SummarizedExperiment and MultiAssayExperiment but possibly others. OMA is not a book about general statistics (a topic which has more comprehensive treatments elsewhere). A key shortcoming in this example is that it does not show how to do regression on such data objects.

Another gap is in the statistical assumptions; read counts or relative abundances in microbiome context usually violate assumptions of standard linear regresssion in multiple ways and that is pedagogically not ideal. Examples on GLMs would be better justified but for those we do have DA tests already available for individual taxa.

If we keep linear regression example then I would implement following changes:

- use one of the microbiome demo data sets
- regress on alpha diversity (Shannon index?) instead of individual taxa (as we already have more dedicated DA methods available for taxa)
- emphasize linear modeling assumptions and include some testing on how well the assumptions hold
- show how to treat possible covariates
- emphasize visualization and interpretation of the analysis outcomes
- where possible, use visualization methods that are dedicated to (Tree)SE/SCE objects, e.g. from miaViz, scater or elsewhere

antagomir committed:
Merge branch 'devel' into jkc9886-patch-8

jkc9886 commented:
I am working on your suggested changes but I see the PR has been approved to merge, is it a technical error or should I continue with adding the changes?

antagomir committed:
Merge branch 'devel' into jkc9886-patch-8

antagomir commented:
This PR has not been merged. 

If you check those "merge" announcements above you can see that they are instead synchronizing this PR with the other PRs that have been approved meanwhile. So the changes from other PRs in the devel branch are merged into your branch to keep it up-to-date, but your branch has not been merged yet to devel branch..

jkc9886 commented:
Oh I see, i get it now, thank you Professor!

jkc9886 committed:
Update

1. Used the GlobalPatterns dataset from mia & compared against Shannon Index instead of individual taxa method. 
2. Checked assumptions such as linearity, normality, homoscedasticity, and independence of residuals.
3. Include sex as a secondary covariate in the regression model.
4. Used scater for PCA and visualization.
5. Added violin plot as well, for extra reference.

antagomir committed:
Merge branch 'devel' into jkc9886-patch-8

jkc9886 committed:
Merge branch 'devel' into jkc9886-patch-8

antagomir committed:
Merge branch 'devel' into jkc9886-patch-8

antagomir commented on inst/pages/exercises.qmd:
````diff
@@ -1741,6 +1741,188 @@ pheatmap(
     annotation_col = as.data.frame(sample_metadata))
 ```
 
+### *Bonus* Regression Charts to Visualize Microbiome Data :
````
Two stylistic updates to all titles:
- let us not capitalize the words in titles (except the first word)
- no double period (:) in the end of a title

antagomir commented on inst/pages/exercises.qmd:
```diff
@@ -1741,6 +1741,188 @@ pheatmap(
+Let us do a regression analysis exercise using the `GlobalPatterns` dataset
+with the following variables:
+
+1. ShannonIndex
```
You could write here: 
- Shannon index ("ShannonIndex")
-

antagomir commented on inst/pages/exercises.qmd:
```diff
@@ -1741,6 +1741,188 @@ pheatmap(
+6. Test the linear modeling assumptions.
+7. Treat other possible covariates.
+8. Perform visualizations and interpret the analysis outcomes. You can use
+visualization methods from packages like miaViz, scater, etc.
```
If you refer to these packages, add a hyperlink to their websites

antagomir commented on inst/pages/exercises.qmd:
```diff
@@ -1741,6 +1741,188 @@ pheatmap(
+tse <- GlobalPatterns
+
+# Calculate the Shannon diversity index
+diversity <- estimateDiversity(tse, assay.type = "counts", index = "shannon")
```
This is now `addAlpha()` in the latest mia version, please update

antagomir replied:
And what comes out is the same tse object, with updated colData. Probably you do not want to change the object name. The idea is that we have one object (tse) and then we can just add material in it.

jkc9886 replied:
I see, got your point.

antagomir commented on inst/pages/exercises.qmd:
```diff
@@ -1741,6 +1741,188 @@ pheatmap(
+data <- data.frame(
+  Shannon_Index = shannon_index,
+  BMI_Group = bmi_group
+)
```
Not necessary. You get data frame (if you need) with just:

`df <- colData(tse)[, c("shannon", "SampleType")]`

However I do not think this is necessary. See next comment.

antagomir commented on inst/pages/exercises.qmd:
```diff
@@ -1741,6 +1741,188 @@ pheatmap(
+)
+
+# Perform the regression analysis
+model <- lm(Shannon_Index ~ BMI_Group, data = data)
```
It doesn't matter if your data has other variables than those you use in the modeling. I think this should work just like:

`lm(shannon~SampleType, data=colData(tse))`

or at least

`lm(shannon~SampleType, data=as.data.frame(colData(tse)))`

or 

`lm(tse$shannon~tse$SampleType)`

antagomir replied:
It makes code unnecessarily longer and more complicated to follow if you unnecessarily create new data frames and rename variables etc. Just use the original data object directly when possible.

antagomir commented on inst/pages/exercises.qmd:
```diff
@@ -1741,6 +1741,188 @@ pheatmap(
+model <- lm(Shannon_Index ~ BMI_Group, data = data)
+
+# Create the plot with the regression line
+p <- ggplot(data, aes(x = BMI_Group, y = Shannon_Index)) +
```
Whynot just 

`ggplot(tse, aes(x=SampleType, y=shannon))`

antagomir commented on inst/pages/exercises.qmd:
```diff
@@ -1741,6 +1741,188 @@ pheatmap(
+# Create the plot with the regression line
+p <- ggplot(data, aes(x = BMI_Group, y = Shannon_Index)) +
+  geom_point(color = 'blue') +
+  geom_smooth(method = 'lm', color = 'red') +
```
geom_smooth() creates its own linear model, you do not need the previous lm() command for this ggplot visualization. It would be good to remove either one of these, or explain in the text that geom_smooth() recalculates the lm for visualization

antagomir commented on inst/pages/exercises.qmd:
```diff
@@ -1741,6 +1741,188 @@ pheatmap(
+
+# Extract the Shannon diversity index and bmi_group
+shannon_index <- diversity$shannon
+bmi_group <- colData(tse)$SampleType  # Using SampleType as a proxy for BMI_Group
```
Sample types in this data set do not seem to have anything to do with BMI? Have you checked the data and examples thoughtfully?


 table(GlobalPatterns$SampleType)

             Feces         Freshwater Freshwater (creek)               Mock 
                 4                  2                  3                  3 
             Ocean Sediment (estuary)               Skin               Soil 
                 3                  3                  3                  3 
            Tongue 
                 2

antagomir commented on inst/pages/exercises.qmd:
```diff
@@ -1741,6 +1741,188 @@ pheatmap(
+print(p)
+
+# Print the model summary
+summary(model)
```
You could here give examples of summarizing table with [gtsummary](https://www.danieldsjoberg.com/gtsummary/) and visualizing with [jtools](https://www.jtools.jacob-long.com/articles/summ.html)

antagomir replied:
Good to cite those package websites in the text explanation then also.

jkc9886 replied:
Sure, did that :))

antagomir replied:
resolve conversations when dealt with

antagomir replied:
resolve conversations when dealt with

antagomir commented on inst/pages/exercises.qmd:
```diff
@@ -1741,6 +1741,188 @@ pheatmap(
+What do we interpret from the above regression analysis?
+
+1. The coefficients of the regression model indicate the relationship 
+between the BMI groups and the Shannon diversity index. If the coefficient
```
are these really BMI groups?

antagomir commented on inst/pages/exercises.qmd:
```diff
@@ -1741,6 +1741,188 @@ pheatmap(
+for a particular BMI group is significantly different from zero, it
+suggests that this group has a different Shannon diversity index compared
+to the reference group.
+2. The p-value associated with each coefficient tells us whether the effect
+of that BMI group on the Shannon diversity index is statistically
+significant. A common threshold for significance is 0.05.
```
Does the lm() output include multiple testing correction?

antagomir commented on inst/pages/exercises.qmd:
````diff
@@ -1741,6 +1741,188 @@ pheatmap(
+
+Now, let's check some linear modeling assumptions :
+
+```{r}
+#| label: sub-checking-modeling-assumptions
+
+# 1. Linearity
+plot(model, which = 1)
+
+# 2. Normality of residuals
+plot(model, which = 2)
+
+# 3. Homoscedasticity
+plot(model, which = 3)
+
+# 4. Independence of residuals (Durbin-Watson test)
+dwtest(model)
+```
+Here, 
+1. Linearity plot(Plot 1) shows the residuals vs. fitted values. The residuals
+should be randomly scattered around zero, indicating that the linearity
````
Either explain each of these as separate code examples, or organize all subplots on a single row.

antagomir commented on inst/pages/exercises.qmd:
````diff
@@ -1741,6 +1741,188 @@ pheatmap(
+```{r}
+#| label: sub-trying-possible-covariates
+
+sex <- colData(tse)$SampleType
+
+# Add sex to the data frame
+data$Sex <- sex
+
````
again, no need to create new variables as you can directly run models with original data

antagomir commented on inst/pages/exercises.qmd:
```diff
@@ -1741,6 +1741,188 @@ pheatmap(
+#| label: sub-scater-visualization
+
+# Plot PCA for Shannon index
+sce <- SingleCellExperiment(assays = list(counts = assay(tse, "counts")),
+colData = colData(tse))
+
```
There is no need to create a new data object. Your tse object already is a SCE object (all TreeSE are SCE).

antagomir replied:
can you just follow the examples on PCA plotting that are available in OMA for tse objects?

antagomir replied:
and is this PCA visualization necessary - what is the relation with the lm model? (we also have other exercises on PCA I think).

jkc9886 replied:
Yes this is not necessarily relevant, i made the corrections.

antagomir commented on inst/pages/exercises.qmd:
```diff
@@ -1741,6 +1741,188 @@ pheatmap(
+sce <- runPCA(sce, ncomponents = 2, exprs_values = "counts")
+
+# Plot PCA with Shannon Index colored by BMI Group
+plotPCA(sce, shape_by = "SampleType") +
+  labs(title = 'PCA of Shannon Index colored by BMI Group')
```
SampleType is not a BMI group

antagomir commented on inst/pages/exercises.qmd:
````diff
@@ -1741,6 +1741,188 @@ pheatmap(
+```{r}
+#| label: sub-violin_plot-visualization
+
+violin_plot <- ggplot(data, aes(x = BMI_Group, y = Shannon_Index,
+fill = BMI_Group)) +
+  geom_violin() +
+  geom_jitter(width = 0.2) +
+  labs(title = 'Violin Plot of Shannon Index vs. BMI Group', x = 'BMI Group',
+  y = 'Shannon Index') +
+  theme_minimal()
+
+# Print the violin plot
+print(violin_plot)
+
+# Visualize the regression results with a summary plot
+par(mfrow=c(2,2))  # Setting up a 2x2 plot grid
+plot(model_with_covariates)
+```
````
You have violin plots readily available for tse objects in miaViz and should use those. Ask Muluh if you don't find otherwise.

antagomir reviewed (changes requested):
See the comments. 

THings to improve:

- still often using base R rather than the TreeSE specific tools (I suggested some changes on this)

- lm models can be used with both continuous data (like continuous age or bmi) and groups but the interpretation is different. It would be best to also include example with a continuous variable because that is a different case. The ggplot visualization with geom_smooth() is usually done with continuous variables, and linking it here with the discrete group variable example is potentially a bit misleading. Show boxplots (using miaViz) instead for that example, and scatterplots (geom_point & geom_smooth) with the continuous variable.

antagomir commented:
There is one major comment related to the use of lm:

1) Your current use case with discrete x, continuous y can be done but is possibly a bit less standard than simple x,y scatterplot with continuous x. For this, scatterplot and geom_smooth are not recommended visualizations as they are designed for continouous x. Use instead boxplots or violin plots to visualize this kind of data (with discrete x).

2) Use with continuous variables (both x, y) is common use case and can be visualized with scatterplots & geom_smooth; this is a common use case and for clarity it would be good to also include example on this. Then the coefficients have different interpretation.

jkc9886 committed:
Update exercises.qmd

1. Introduced two examples : one with discrete x and one with continuous x against shannon index.
2. Made changes to the steps and interpretations accordingly while adding gtsummary and jtools for summary and coefficient visualization.

jkc9886 committed:
Update exercises.qmd

Removing the unused libraries from line 1796

antagomir commented on inst/pages/exercises.qmd:
```diff
@@ -1741,6 +1741,175 @@ pheatmap(
+continuous variable from the dataset’s metadata. These variables will be used
+as predictors/independent variables in our linear regression models.
+
+Steps:
+1. Import the dataset into R; here we are using a microbiome dataset.
+2. Calculate Alpha Diversity (Shannon Index), i.e. the dependent variable and 
+define the covariates, the independent variable.
+3. Fit linear model using discrete variable "SampleType", where Shannon diversity
+is the response variable, and SampleType is the predictor. 
+4. Visualize the results using a boxplot. This plot helps in understanding the
+distribution of diversity across discrete sample types.
+5. Fit linear model using continuous variable "coverage", where Shannon diversity 
+is the response variable, and coverage is the predictor.  
+6. Visualize the model with a scatterplot. In R, you can also use 
+[`ggplot2`](https://ggplot2.tidyverse.org/) for
+plotting and `lm()` function for linear modeling.
+7. Examine the slope, intercept, and R-squared value to understand the 
+relationship between the variables.
+8. Summarize results with `gtsummary`. The [`gtsummary`]
+(https://www.danieldsjoberg.com/gtsummary/) package is used to generate a summary 
+table of the regression model, which is easy to interpret and present in reports
+or publications.
+9. Visualize coefficients with `jtools`. The [`jtools`]
+(https://cran.r-project.org/web/packages/jtools/index.html) package provides a visual
+representation of the model’s coefficients, including their confidence intervals, allowing
+for a clear understanding of the effect sizes.
+10. Finally, interpret the analysis outcomes. 
+
+Visualization methods from packages like 
```
Better to show the code step-by-step?

jkc9886 replied:
I have added relevant comments in the code to show the steps too, do you think I should merge the steps into comments?

jkc9886 replied:
Plus, this gives some space in between the code to talk about the interpretations.

antagomir replied:
Yes, split the code into multiple chunks and explain more per each step.

antagomir commented on inst/pages/exercises.qmd:
```diff
@@ -1741,6 +1741,175 @@ pheatmap(
+
+# Calculate alpha diversity (ShannonIndex)
+tse <- addAlpha(tse, method = "shannon")
+shannon <- colData(tse)$shannon
```
Why would you pick "shannon" here separately? You can call it directly from colData as there is a field called "shannon". The lm method works. Remove this line.

antagomir commented on inst/pages/exercises.qmd:
```diff
@@ -1741,6 +1741,175 @@ pheatmap(
+and the statistical tests), you can be confident that the linear regression model
+is appropriate for the data, and the results can be interpreted reliably.
+
+Further, let's explore a continuous covariate, "coverage".
```
What does this variable represent?

jkc9886 replied:
It represents the sampling depth, or simply how comprehensively the microbial diversity was sampled in each sample. I also thought to share more about the variable (update :  added that in my upcoming commit)

jkc9886 committed:
Update exercises.qmd

Added description to the variable, coverage.
Edited the shannon method command.

jkc9886 committed:
Step by step description.qmd

antagomir committed:
Merge branch 'devel' into jkc9886-patch-8

antagomir commented:
@jkc9886 is this ready from your side?

jkc9886 commented:
Yes Professor.

antagomir committed:
Merge branch 'devel' into jkc9886-patch-8

antagomir commented on inst/pages/exercises.qmd:
```diff
@@ -1741,6 +1741,228 @@ pheatmap(
+#| eval: false
+
+tse <- addAlpha(tse)
+shannon <- colData(tse)$shannon
```
This line is still not removed (see the previous comment)?

antagomir commented on inst/pages/exercises.qmd:
```diff
@@ -1741,6 +1741,228 @@ pheatmap(
+#| label: extra-regressionchart-step3
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: false
```
"eval: true" should be default - is there a reason to not evaluate this? check the same for other chunks

antagomir commented on inst/pages/exercises.qmd:
```diff
@@ -1741,6 +1741,228 @@ pheatmap(
+distance) are potential outliers that may have a strong influence on the model's 
+coefficients.
+
+ii. Durbin-Watson test checks for autocorrelation in the residuals. A value
```
Hmm why "11." comes after "1." and then after this comes "5." -> the numbers do not appear in a logical order?

jkc9886 replied:
I think you mean the ii here ? I used roman numerals for the subsections. So after section 4 which contains subsections i,ii, comes section numbered 5.

antagomir commented on inst/pages/exercises.qmd:
```diff
@@ -1741,6 +1741,228 @@ pheatmap(
+5. Further, fit linear model using a continuous variable, say "coverage", where
+Shannon diversity is the response variable, and coverage is the predictor. 
+
+"Coverage" is a measure of how comprehensively the microbial diversity
+was sampled in each sample. This could refer to the sequencing depth (number of reads)
+or an index like Good’s Coverage that quantifies the completeness of sampling.
```
What is the motivation for comparing Shannon and coverage? This is a technical comparison where some association would be expected. Do we have any biologically relevant comparison available?

jkc9886 replied:
including this comparison can highlight samples that need further sequencing, improving the quality of our data. We risk misinterpreting samples with high diversity but low coverage, where the sequencing depth might not have been sufficient to reveal the full diversity. It shows room to reduce bias but you are right this is an indirect relevance here.

jkc9886 replied:
Also, I am extremely sorry for the delay in my response, my classes have started and keep me occupied but I shall resolve these soon.

antagomir reviewed (changes requested):
Set chunks to eval: true and check the other comments.

jkc9886 committed:
Update exercises.qmd

eval : true, minor edits

0xMuluh committed:
Merge branch 'devel' into jkc9886-patch-8

0xMuluh committed:
Merge branch 'devel' into jkc9886-patch-8

TuomasBorman committed:
Merge branch 'devel' into jkc9886-patch-8

TuomasBorman reviewed (changes requested):
This looks nice but there are couple of problems:

1. It think the instructions in the exercises should be in a format "a) Do this. b) Do that."; with clear instructions what reader should do. This current version at some points explains the reasoning but does not give instructions to user; it just shows how to do it in code chunk.

2. Most of the explanations should be in main material of OMA. The info might be hard to find here.

3. The exercises should reflect the material in the book. In theory, user should be able to solve most of the exercises by copy-pasting code from OMA.

antagomir committed:
Merge branch 'devel' into jkc9886-patch-8

antagomir commented:
Hi @jkc9886 - this PR is still unmerged. Would you help us to finalize this task you initiated?

jkc9886 commented:
Yes, sure Professor. I shall work on the comments suggested by Tuomas and
send the edits.

On Sun, Nov 10, 2024 at 11:26 PM Leo Lahti ***@***.***> wrote:

> Hi @jkc9886
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_jkc9886&d=DwMCaQ&c=slrrB7dE8n7gBJbeO0g-IQ&r=fvCQdMIyOhlZ3JXGliOTRQ&m=YC1zQpLhXKKo-ffDr1EJrYp_-1btb5wJ814atDwHJgRy3z3rOQePW2ia38hFsbw6&s=lrb9xsxN0a0bXYe2REjW_ZJ4qzBZrsqG_W-qqUPsj0k&e=>
> - this PR is still unmerged. Would you help us to finalize this task you
> initiated?
>
> —
> Reply to this email directly, view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_microbiome_OMA_pull_591-23issuecomment-2D2466855485&d=DwMCaQ&c=slrrB7dE8n7gBJbeO0g-IQ&r=fvCQdMIyOhlZ3JXGliOTRQ&m=YC1zQpLhXKKo-ffDr1EJrYp_-1btb5wJ814atDwHJgRy3z3rOQePW2ia38hFsbw6&s=JyttnDscqTcAS9V2dC37-_WgD5H3-YDn8NoNFFI1h28&e=>,
> or unsubscribe
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_A7UNPARCAPNGFXFRFAQ4UULZ76XP5AVCNFSM6AAAAABL4PZEIGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDINRWHA2TKNBYGU&d=DwMCaQ&c=slrrB7dE8n7gBJbeO0g-IQ&r=fvCQdMIyOhlZ3JXGliOTRQ&m=YC1zQpLhXKKo-ffDr1EJrYp_-1btb5wJ814atDwHJgRy3z3rOQePW2ia38hFsbw6&s=zepbJDfwNBVtFhQA2mvomL0fD8taML2_WHUV31SFDcc&e=>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

antagomir committed:
Merge branch 'devel' into jkc9886-patch-8

antagomir committed:
Merge branch 'devel' into jkc9886-patch-8

jkc9886 commented:
Hey @TuomasBorman, thank you for your comments. This is a bonus section so I added the info with it too but do you suggest moving the reasoning/detailed information to any other chapter as a section or is it fine to completely remove the explanation and just provide instructions and code snippets ?

TuomasBorman commented on inst/pages/exercises.qmd:
```diff
@@ -1805,6 +1805,227 @@ mat <- assay(tse, "standardize")
+#| code-summary: "Show solution"
+#| eval: true
+
+model <- lm(shannon ~ SampleType, data = colData(tse))
```
This line does not work, since it does not have "_diversity" suffix

TuomasBorman replied:
You can also modify the line 1853

tse <- addAlpha(tse, index = "shannon")

TuomasBorman commented on inst/pages/exercises.qmd:
```diff
@@ -1805,6 +1805,227 @@ mat <- assay(tse, "standardize")
+#| code-summary: "Show solution"
+#| eval: true
+
+plotColData(tse, x = "SampleType", y = "shannon", color_by = "SampleType") +
```
This should be

plotColData(tse, x = "SampleType", y = "shannon", color_by = "SampleType", show_boxplot = TRUE)

TuomasBorman commented on inst/pages/exercises.qmd:
```diff
@@ -1805,6 +1805,227 @@ mat <- assay(tse, "standardize")
+#| label: extra-regressionchart-step1
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: true
```
Exercise chunks are not run. So this should be eval: false

TuomasBorman commented on inst/pages/exercises.qmd:
```diff
@@ -1805,6 +1805,227 @@ mat <- assay(tse, "standardize")
+
+What do we interpret from the above regression analysis?
+
+i. The boxplot generated shows the distribution of Shannon diversity across
+different sample types. If the boxes are well-separated and there are 
+significant differences in medians, it indicates that SampleType has a strong
+effect on diversity. This would be supported by significant p-values in the
+regression summary.
+ii. The p-value associated with each coefficient tells us whether the effect
+of that sample type on the Shannon diversity index is statistically
+significant. A common threshold for significance is 0.05.
+iii. R-squared: This value indicates the proportion of the variance in the
+Shannon diversity index that is explained by the sample type. A higher R-squared
+value indicates a better fit of the model.
```
Instead of i) ii) ..., use 1, 2, 3

TuomasBorman replied:
The code does not show p-values, you could run

summary(model)

TuomasBorman commented on inst/pages/exercises.qmd:
```diff
@@ -1805,6 +1805,227 @@ mat <- assay(tse, "standardize")
+#| code-summary: "Show solution"
+#| eval: true
+
+model_continuous <- lm(shannon ~ coverage, data = colData(tse))
```
There is no "coverare" and "shannon" indices in colData

TuomasBorman commented on inst/pages/exercises.qmd:
```diff
@@ -1805,6 +1805,227 @@ mat <- assay(tse, "standardize")
+#| code-summary: "Show solution"
+#| eval: true
+
+gt_model <- tbl_regression(model)
```
This is not the model for shannon vs coverage

TuomasBorman commented on inst/pages/exercises.qmd:
```diff
@@ -1805,6 +1805,227 @@ mat <- assay(tse, "standardize")
+
+library(mia)             
+library(TreeSummarizedExperiment) 
+library(scater) 
+library(lmtest)
+library(gtsummary)   
+library(jtools)   
```
Check which packages are really needed

TuomasBorman commented on inst/pages/exercises.qmd:
```diff
@@ -1805,6 +1805,227 @@ mat <- assay(tse, "standardize")
+#| eval: true
+
+library(mia)             
+library(TreeSummarizedExperiment) 
```
TreeSE comes with mia

TuomasBorman commented on inst/pages/exercises.qmd:
```diff
@@ -1805,6 +1805,227 @@ mat <- assay(tse, "standardize")
+Shannon diversity index that is explained by the sample type. A higher R-squared
+value indicates a better fit of the model.
+
+Now, let's check some linear modeling assumptions :
```
Remove space: "modeling assumptions:"

TuomasBorman commented on inst/pages/exercises.qmd:
```diff
@@ -1805,6 +1805,227 @@ mat <- assay(tse, "standardize")
+distance) are potential outliers that may have a strong influence on the model's 
+coefficients.
+
+ii. Durbin-Watson test checks for autocorrelation in the residuals. A value
```
I think this Durbin-Watson test can be removed

TuomasBorman reviewed (commented):
I suggest to:

1.

We can keep the explanation here. I am not sure where these would fit in chapters.

2.

You should rephrase the instructions. Remember that these are exercises, not tutorial. User should be able to follow without checking the solution.

> First, fit linear model using a discrete variable, say "SampleType", where  Shannon diversity is the response variable, and SampleType is the predictor.

You could say that user can use base R functions to fit linear model.

--> ".... (see `stats::lm()`)"

> Now, let's check some linear modeling assumptions :

How user should do that? It is not said anywhere?

--> "You can plot the fitted model (hint: plot())

3.

To make this exercise simpler, you could focus only one case, e.g., shannon vs sample type

4.

You could use some other dataset or use different grouping (e.g., human vs environment). Now the groups have 2 samples per group.

5.
Check that these examples run with the latest mia

TuomasBorman commented on inst/pages/exercises.qmd:
````diff
@@ -1805,6 +1805,227 @@ mat <- assay(tse, "standardize")
+print(dw_result)
+```
+
+Here, 
+
+i. Residual diagnostics (normality and homoscedasticity):
+
+a. The Residuals vs Fitted plot provides insight into the relationship 
+between residuals and fitted values. Ideally, the residuals should be randomly
+scattered around the horizontal line (residuals = 0) without any discernible 
+pattern. This indicates that the model’s assumption of linearity and homoscedasticity
+(constant variance of residuals) is likely met. If you observe any systematic patterns
+(such as a funnel shape), it may suggest heteroscedasticity, meaning the variance of
+residuals is not constant.
````
You could add question: "Do the assumptions hold?". And then add answer with folded text: https://quarto.org/docs/authoring/callouts.html

TuomasBorman commented:
Any updates?

TuomasBorman commented:
Closing as these are outdated.
